app/controllers/tasks_controller.rb

```diff
@@ -15,6 +15,15 @@ def index
     @tasks = @search.result(distinct: true).paginate(per_page: 5, page:)
   end
 
+  def duplicate
+    new_entry = @task.clean_duplicate(current_user)
+    if new_entry.save(context: :force_validations)
+      redirect_to new_entry, notice: t('tasks.notification.created')
+    else
+      redirect_to @task, alert: t('tasks.notification.duplicate_failed')
```

Comment:
This case should not be able to happen, however I don't know how to test this.

The issue is, that you would need to create an invalid task to duplicate it, which seems to be impossible in RSpec.

Reply:
As discussed, you might be able to test it with a mock, such as

```rb
let(:invalid_task) { Task.new }
// ...
allow(task).to receive(:clean_duplicate).with(user).and_return(invalid_task)
```

Reply:
For some reason, this does not work as expected.

I have the option to use `any_instance_of(Task)`, but rubocop does not like that solution.

Do you have an alternative idea?

Reply:
Indeed, it doesn't work as expected. The main reason is that we are specifically stubbing one method on one object. However, the controller will create a new object from the database (mirroring the same underlying task), but hence our stubbing no longer works. We can, however, instruct the controller to find exactly the same object:

```rb
context 'when saving fails' do
  let(:invalid_task) { Task.new }
  
  before do
    # We need to stub the find method, because otherwise another object is returned from the database
    allow(Task).to receive(:find).with(task.id.to_s).and_return(task)
    # By design, we return an invalid task, which will fail to save
    allow(task).to receive(:clean_duplicate).with(user).and_return(Task.new)
  end

  it 'shows an error message' do
    post_duplicate
    expect(flash[:alert]).to eq(I18n.t('tasks.notification.duplicate_failed'))
  end
end
```

Reply:
It makes sense. Thank you for your help.